src/test/unit_tests/test_namespace_fs.js

```diff
@@ -1764,8 +1764,111 @@ mocha.describe('namespace_fs copy object', function() {
         });
     });
 
+    mocha.describe('namespace_fs upload object with tagging', function() {
+
+        const upload_bkt_tagging = 'test_ns_uploads_object_tagging';
+        const tmp_fs_path_tagging = path.join(TMP_PATH, 'test_namespace_fs_tagging');
+        const ns_tmp_bucket_path_tagging = `${tmp_fs_path}/${src_bkt}`;
+        const ns_tmp_tagging = new NamespaceFS({ bucket_path: ns_tmp_bucket_path_tagging, bucket_id: '3', namespace_resource_id: undefined });
+        mocha.before(async () => {
+            await P.all(_.map([src_bkt, upload_bkt_tagging], async buck =>
+                fs_utils.create_fresh_path(`${tmp_fs_path_tagging}/${buck}`)));
+        });
+        mocha.after(async () => {
+            await P.all(_.map([src_bkt, upload_bkt_tagging], async buck =>
+                fs_utils.folder_delete(`${tmp_fs_path_tagging}/${buck}`)));
+        });
+        mocha.after(async () => fs_utils.folder_delete(tmp_fs_path_tagging));
+        mocha.describe('upload_object (tagging)', function() {
+            const upload_key = 'upload_key_1';
+            const upload_folder_key = 'upload_folder_key/';
+            const upload_folder_key_without_slash = 'upload_folder_key';
+            const data = crypto.randomBytes(100);
+            const tagging_1 = [{ key: 'tag1', value: 'val1' }];
+            const xattr = { key: 'key1', key2: 'value1' };
+
+            mocha.it('Upload object with tagging', async function() {
+                const params = {
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                    xattr,
+                    source_stream: buffer_utils.buffer_to_read_stream(data),
+                    tagging: tagging_1,
+                };
+                const upload_res = await ns_tmp_tagging.upload_object(params, dummy_object_sdk);
+                console.log('upload_object (tag) response', inspect(upload_res));
+                const tag_res = await ns_tmp_tagging.get_object_tagging({
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                }, dummy_object_sdk);
+                console.log('get_object_tagging response', inspect(tag_res));
+                await validate_tagging(tag_res, tagging_1);
+            });
+
+            mocha.it('Upload object with tagging, update object and tag', async function() {
```

Comment:
Could you please explain where is the part of "update object" (I see only the "update tag" where it is `tagging_2` instead of `tagging_1`)?